_ingest-pipelines/processors/fail.md

```diff
@@ -0,0 +1,154 @@
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline.
+
+### Step 1: Create a pipeline
+
+The following query creates a pipeline, named `fail-log-pipeline`, that uses the `fail` processor to intentionally fail the pipeline execution for log events: 
+
+```json
+PUT _ingest/pipeline/fail-log-pipeline  
+{  
+  "description": "A pipeline to test the fail processor for log events",  
+  "processors": [  
+    {  
+      "fail": {  
+        "if": "ctx.user_info.contains('password') || ctx.user_info.contains('credit card')",  
+        "message": "Document containing personally identifiable information (PII) cannot be indexed!"  
+      }  
+    }  
+  ]  
+}
+```
+{% include copy-curl.html %}
+
+### Step 2 (Optional): Test the pipeline
+
+It is recommended that you test your pipeline before you ingest documents.
+{: .tip}
+
+To test the pipeline, run the following query:
+
+```json
+POST _ingest/pipeline/fail-log-pipeline/_simulate  
+{  
+  "docs": [  
+    {  
+      "_source": {  
+        "user_info": "Sensitive information including credit card"  
+      }  
+    }  
+  ]  
+}  
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following example response confirms that the pipeline is working as expected:
+
+```json
+{
+  "docs": [
+    {
+      "error": {
+        "root_cause": [
+          {
+            "type": "fail_processor_exception",
+            "reason": "Document containing personally identifiable information (PII) cannot be indexed!"
+          }
+        ],
+        "type": "fail_processor_exception",
+        "reason": "Document containing personally identifiable information (PII) cannot be indexed!"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Step 3: Ingest a document 
+
+The following query ingests a document into an index named `testindex1`:
+
+```json
+PUT testindex1/_doc/1?pipeline=fail-log-pipeline  
+{  
+  "user_info": "Sensitive information including credit card"  
+} 
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The request fails to index the log event into the index `testindex1` due to the string `credit card` being present in `user_info`. The following response includes the custom error message specified in the fail processor:
+
+```json
+
+  "error": {
+    "root_cause": [
+      {
+        "type": "fail_processor_exception",
+        "reason": "Document containing personally identifiable information (PII) cannot be indexed!"
+      }
+    ],
+    "type": "fail_processor_exception",
+    "reason": "Document containing personally identifiable information (PII) cannot be indexed!"
+  },
+  "status": 500
+}
+```
+{% include copy-curl.html %}
+
+### Step 4 (Optional): Retrieve the document
+
+Since the log event was not indexed due to the pipeline failure, attempting to retrieve it will result in a document not found error.
```

Comment:
```suggestion
Because the log event was not indexed due to the pipeline failure, attempting to retrieve it will result in a document not found error:
```